code/modules/mob/living/carbon/human/MedicalSideEffects.dm

```diff
@@ -40,23 +39,21 @@
 			M.start = life_tick
 			return
 
-
-	var/T = side_effects[name]
-	if (!T)
-		return
-
-	var/datum/medical_effect/M = new T
-	if(M.name == name)
-		M.strength = strength
-		M.start = life_tick
-		side_effects += M
+	var/list/L = subtypesof(/datum/medical_effect)
+	for(var/T in L)
+		var/datum/medical_effect/M = new T
+		if(M.name == name)
+			M.strength = strength
+			M.start = life_tick
+			side_effects += M
+			break
 
 /mob/living/carbon/human/proc/handle_medical_side_effects()
 	//Going to handle those things only every few ticks.
 	if(life_tick % 15 != 0)
-		return 0
+		return FALSE
 
-	var/list/L = typesof(/datum/medical_effect)-/datum/medical_effect
+	var/list/L = subtypesof(/datum/medical_effect)
 	for(var/T in L)
 		var/datum/medical_effect/M = new T
```

Comment:
As above, these need to use `initial()` or clean up after being instantiated. Preferrably they should be cached somewhere for reference.

Reply:
If you just add `else qdel(M)` at line 64 this will be ok.